lib/matplotlib/textpath.py

```diff
@@ -243,25 +242,28 @@ def get_glyphs_tex(self, prop, s, glyph_map=None,
 
         # Gather font information and do some setup for combining
         # characters into strings.
-        for x1, y1, dvifont, glyph, width in page.text:
-            font, enc = self._get_ps_font_and_encoding(dvifont.texname)
-            char_id = self._get_char_id(font, glyph)
-
+        for text in page.text:
+            font = get_font(text.font_path)
+            char_id = self._get_char_id(font, text.glyph)
             if char_id not in glyph_map:
+                spec = text.glyph_spec
                 font.clear()
                 font.set_size(self.FONT_SCALE, self.DPI)
-                # See comments in _get_ps_font_and_encoding.
-                if enc is not None:
-                    index = font.get_name_index(enc[glyph])
+                # See comments in Text.get_glyph_spec.
```

Comment:
If you really want to stick with the type instability, I suggest to do
 ```
glyph_name_or_index = text.glyph_spec
if isinstance(glyph_name_or_index, str):  # glyph name
    ...
elif isinstance(glyph_name_or_index, int):  # glyph index
    ...
```
Maybe even call the property `glyph_name_or_index`.

Reply:
Done.